backend/clubs/views.py

```diff
@@ -2979,6 +2981,22 @@ def get_serializer_class(self):
         return UserProfileSerializer
 
 
+class ClubApplicationViewSet(viewsets.ModelViewSet):
+    """
+    list: Retrieve a list of applications of the club.
+
+    get: Retrieve the details for a given application.
+    """
+
+    permission_classes = [IsSuperuser]
```

Comment:
You don't want to just do `IsSuperuser`, since normal users need to view the applications and club officers need to be able to create, edit, and delete the applications.

Look at `ClubItemPermission` for more details, and how the `TestimonialViewSet` and the `EventViewSet` use it.